Sources/Sentry/SentryTracer.m

```diff
@@ -607,7 +601,24 @@ - (void)finishInternal
 
 #if SENTRY_TARGET_PROFILING_SUPPORTED
     if (self.isProfiling) {
-        [self captureTransactionWithProfile:transaction];
+        NSDate *startTimestamp;
+
+#    if SENTRY_HAS_UIKIT
+        if (appStartMeasurement != nil) {
+            startTimestamp = appStartMeasurement.runtimeInitTimestamp;
+        }
+#    endif // SENTRY_HAS_UIKIT
+
+        if (startTimestamp == nil) {
+            startTimestamp = self.startTimestamp;
+        }
+        if (!SENTRY_ASSERT_RETURN(startTimestamp != nil,
+                @"A transaction with a profile should have a start timestamp already. We will "
+                @"assign the current time but this will be incorrect.")) {
+            startTimestamp = [SentryDependencyContainer.sharedInstance.dateProvider date];
```

Comment:
m: Does it make sense if its incorrect?

Reply:
I agree; maybe it makes sense to ditch the profile in that case instead of using an incorrect time.

Reply:
It doesn't necessarily make sense, but this is how we previously handled it. It's just that the same decision had to move to this new location in code. I would say let's not worry about it because this will not matter once we move to continuous profiling: we won't have to deal with the potentially nil timestamp from any txn, we will be able to just use our own.